### PR TITLE
3-calendar-view-for-month

### LINKED ISSUE #3: Calendar view for month
# What is the Problem?
There is no month calendar view.

# Why is This a Problem?
Without a month calendar view, the user cannot see their schedule for the selected month in a glance.

# How do We Fix it?
We'll add a calendar view with 31 blocks for every day of the month up to 31 days.

### PR DESCRIPTION
**Pull request by BetOnJeanette**
# What was the problem?
There was no monthly layout for the calendar

# How does this address the problem?
I added a header that shows the current month and year being shown, with links that move to the previous and next months.

I created a shared component for weekly layout, as well as the shared style to have a body evenly divided between 7 sections.

I also implemented the base layout, where each month has a filler space between sunday and the first day of the month (unless it starts on a sunday). Then, we fill the rest of the month with day components.

This closes #3.

### CODE REVIEW

File: src/CalendarViews/SharedStyling/WeekFormating.css [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.weekFormat {
+    display: grid;
+    grid-template-columns: calc(100% / 7) calc(100% / 7) calc(100% / 7) calc(100% / 7) calc(100% / 7) calc(100% / 7) calc(100% / 7);
```
Comment (MsZoezo) [inline src/CalendarViews/SharedStyling/WeekFormating.css:3]: while this works, there are built in ways to repeatedly assign fractions of the width to a column.

```css
grid-template-columns: repeat(7, 1fr);
```
this is cleaner to read and if you add something like a gap later on it'll handle it no problem too.

**Review by MsZoezo — CHANGES_REQUESTED** [review]
Currently the .tsx files are also quite messy in the use (or lack of) ; at the end of lines.
(Something I seem to have missed in my previous review >~<)

Either way I do think it's important to keep being consistent with this, maybe we should look into a linter after all?

**Comment by BetOnJeanette** [comment]
Updated the branch: 
- Used the CSS repeat (sorry, didn't know that was a thing)
- Pushed semicolons off (see #21)
- Destructured props
- Added JSDoc comments for everything

File: src/CalendarViews/MonthlyCalendar/MainBody/StartOfMonthBuffer/StartOfMonthBuffer.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+/**
+ * Creates a buffer div for the start of the month
+ * @param props The information needed to be passed in
+ * @returns If the first day is a sunday, an empty element. 
+ *          Otherwise, a buffer element between sunday and the first date.
+ */
+function StartOfMonthBuffer(props: Props){
+    let { firstOfMonth } = props
```
Comment (MsZoezo) [inline src/CalendarViews/MonthlyCalendar/MainBody/StartOfMonthBuffer/StartOfMonthBuffer.tsx:12]:
```suggestion
function StartOfMonthBuffer({ firstOfMonth }: Props){
```

I would recommend destructuring like this, keeps it all in one line all neatly.

File: src/CalendarViews/MonthlyCalendar/MainBody/SingleDate/SingleDate.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+interface Props{
+    date: Date
+}
+/**
+ * Gets the element for a single day
+ * @param props The necessary information to be passed in
+ * @returns A JSX element representing the day.
+ */
+function SingleDate(props: Props){
+    let { date } = props
```
Comment (MsZoezo) [inline src/CalendarViews/MonthlyCalendar/MainBody/SingleDate/SingleDate.tsx:10]:
```suggestion
function SingleDate({ date }: Props){
```

File: src/CalendarViews/MonthlyCalendar/MainBody/MainBody.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    date: Date
+}
+
+/**
+ * Creates the main body of a monthly calendar
+ * @param props The props used to recreate the element
+ * @returns A JSX copy of the body of the monthly view
+ */
+function CalendarBody(props: Props) {
+    let {date} = props
```
Comment (MsZoezo) [inline src/CalendarViews/MonthlyCalendar/MainBody/MainBody.tsx:15]:
```suggestion
function CalendarBody({ date }: Props) {
```

File: src/CalendarViews/MonthlyCalendar/Header/CalendarHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    changeMonthYear: (a: number)=> void;
+}
+
+/**
+ * Creates the header for a monthly calendar
+ * @param props The information necessary to create the header
+ * @returns a JSX element for the header 
+ */
+function CalendarHeader(props: Props) {
+    let {date, changeMonthYear} = props
```
Comment (MsZoezo) [inline src/CalendarViews/MonthlyCalendar/Header/CalendarHeader.tsx:14]:
```suggestion
function CalendarHeader({ date, changeMonthYear }: Props) {
```

**Review by MsZoezo — CHANGES_REQUESTED** [review]

**Review by MsZoezo — APPROVED** [review]
This code transcends time (which may be an issue when you deal with a calendar)🌠

**Pull request merged**